File: app/src/main/java/com/tianma/xsmscode/ui/record/CodeRecordActivity.java
package com.tianma.xsmscode.ui.record;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.widget.Toolbar;

import com.github.tianma8023.xposed.smscode.R;
import com.tianma.xsmscode.common.fragment.backpress.BackPressEventDispatchHelper;
import com.tianma.xsmscode.ui.app.base.BaseActivity;

import com.github.tianma8023.xposed.smscode.databinding.ActivityCodeRecordsBinding;

/**
 * Sms Code Records
 */
public class CodeRecordActivity extends BaseActivity {

    private ActivityCodeRecordsBinding binding;

    public static void startToMe(Context context) {
        Intent intent = new Intent(context, CodeRecordActivity.class);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityCodeRecordsBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setupToolbar();

        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.code_records_main_content, CodeRecordFragment.newInstance())
                .commit();
    }

    private void setupToolbar() {
        setSupportActionBar(binding.toolbar.getRoot());
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setHomeButtonEnabled(true);
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }

    @Override
    public void onBackPressed() {
        if (!BackPressEventDispatchHelper.dispatchBackPressedEvent(this)) {
            super.onBackPressed();
        }
    }
}

File: app/src/main/java/com/tianma/xsmscode/ui/record/CodeRecordActivity.kt
package com.tianma.xsmscode.ui.record

import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.github.tianma8023.xposed.smscode.R
import com.github.tianma8023.xposed.smscode.databinding.ActivityCodeRecordsBinding
import com.tianma.xsmscode.ui.app.base.BaseActivity

/**
 * Sms Code Records
 */
class CodeRecordActivity : BaseActivity() {

    private lateinit var binding: ActivityCodeRecordsBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityCodeRecordsBinding.inflate(layoutInflater)
        setContentView(binding.root)
        handleInsets(binding.root)

        setupToolbar()

        supportFragmentManager
            .beginTransaction()
            .replace(R.id.code_records_main_content, CodeRecordFragment.newInstance())
            .commit()
    }

    private fun setupToolbar() {
        setSupportActionBar(binding.toolbar.root)
        supportActionBar?.apply {
            setHomeButtonEnabled(true)
            setDisplayHomeAsUpEnabled(true)
        }
    }


    companion object {
        @JvmStatic
        fun startToMe(context: Context) {
            val intent = Intent(context, CodeRecordActivity::class.java)
            context.startActivity(intent)
        }
    }
}
